### PR TITLE
Add PInvoke/CriticalHandles tests

### PR DESCRIPTION
**Pull request by jiangzeng**

### CODE REVIEW

File: tests/src/Interop/PInvoke/CriticalHandles/Test/Test.cs [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+        IntPtr value;
+        value = Native.In(handle, null);
+        Assert.AreEqual(handleValue.ToInt32(), value.ToInt32(), "Handle value");
+    }
+
+    public static void Ret()
+    {
+        IntPtr handleValue = new IntPtr(2);
+        //TODO: Expected MissingMemberException but throws MissingMethodException
+        Assert.Throws<MissingMethodException>(() => Native.Ret(handleValue), "Calling P/Invoke that returns an no default ctor critical handle");
```
Comment (jiangzeng) [inline tests/src/Interop/PInvoke/CriticalHandles/Test/Test.cs]: @AaronRobinsonMSFT Need confirm here which exception should throw

Comment (AaronRobinsonMSFT) [reply tests/src/Interop/PInvoke/CriticalHandles/Test/Test.cs]: I believe @hughbe was also reporting this in a test in corefx. Can you confirm that @hughbe?

Can you check if we have this coverage in corefx? If so, please remove here.

Comment (jiangzeng) [reply tests/src/Interop/PInvoke/CriticalHandles/Test/Test.cs]: @AaronRobinsonMSFT I did not found similar test in CoreFx., @hughbe do you have any idea?

Comment (hughbe) [reply tests/src/Interop/PInvoke/CriticalHandles/Test/Test.cs]: I didn’t. All I can think of is something to do with handles and getting a variant from them not being allowed

**Comment by jkoritzinsky** [comment]
@dotnet-bot test this please

**Comment by jkoritzinsky** [comment]
@dotnet-bot test OSX10.12 x64 Checked Innerloop Build and Test

**Comment by jkoritzinsky** [comment]
@janvorli I think the ReverseTest in this branch also reproduces the x86 exception unwinding bug.

**Comment by jkoritzinsky** [comment]
@dotnet-bot test OSX10.12 x64 Checked Innerloop Build and Test 
@dotnet-bot test Ubuntu x64 Checked Innerloop Build and Test (Jit - TieredCompilation=0)

**Comment by jkoritzinsky** [comment]
@dotnet-bot test this please

**Pull request merged**